### DISCUSSION TITLE
Deleting default keymap from within plugin file

### DISCUSSION THREAD

**Discussion by kapral18** [LazyVim/LazyVim#1615] [Q&A] [answered]
I'd love to have related configuration colocated, hence I am trying to delete the lazyterm bindings to not get in the way of me hitting the closely located <C-\>

Basically trying to remove default lazyvim <C-/> bindings, but when I try to remove them either in init= block or outside of plugin table I get this error.

<img width="1728" alt="CleanShot 2023-10-07 at 15 54 02@2x" src="https://github.com/LazyVim/LazyVim/assets/1625373/5788150a-7c3a-4f41-9926-54e4407b29e4">

It totally works if I move the keymap.del into lua/keymaps.lua. but I want it colocated with my plugin code.

Is there a way to achieve from within plugin file?

**Comment by folke** [comment]
Nope: https://www.lazyvim.org/configuration/keymaps

**Reply by kapral18** [reply]
I see thanks of pointing out

**Comment by dpetka2001** [comment, marked as answer]
@kapral18 
You could try the following
```lua
    config = function(_, opts)
      require("toggleterm").setup(opts)
      vim.keymap.del({ "n", "t" }, "<C-/>")
    end,
```
but this will only delete the default Lazyvim keymap once the plugin is loaded. 

Also instead of creating 10 keymaps for 5 terminals in horizontal and vertical, you could only create 2 keymaps like so
```lua
      {
        "<leader>Th",
        function()
          local count = vim.v.count1
          require("toggleterm").toggle(count, 10, Util.get_root(), "horizontal")
        end,
        desc = "ToggleTerm (horizontal)",
      },
```
thus taking advantage of the Toggleterm API which uses the `v.count` as parameter to invoke a terminal with a number prefix.

Edit: you can take a look [here](https://github.com/akinsho/toggleterm.nvim/blob/c80844fd52ba76f48fabf83e2b9f9b93273f418d/lua/toggleterm.lua#L278-L284) to understand the parameters it accepts.

**Reply by kapral18** [reply]
thanks for the solution, works perfectly. And thanks for the additional advice, that's indeed way more convenient.
